scripts/cangen/cangen/can_generator.py

```diff
@@ -247,13 +237,25 @@ def generate_code(bus: Bus, config: Config):
     logger.info("Code generation complete")
 
 
+def _prepare_output_directory(output_dir):
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+
+    os.makedirs(output_dir, exist_ok=True)
+
+    # Create a git ignore for everything in the generated path. Ignore everything.
+    gitignore_path = os.path.join(output_dir, ".gitignore")
+    if not os.path.exists(gitignore_path):
+        with open(gitignore_path, "w") as f:
+            f.write("*")
+
+
 def generate_can_from_dbc(project_folder_name: str):
     os.chdir(project_folder_name)
     config = Config.from_yaml("config.yaml")
 
-    # Deletes output path folder and files within, before creating new ones
-    if os.path.exists(config.output_dir):
-        shutil.rmtree(config.output_dir)
+    # Deletes previously generated files and creates a gitignore for the directory
```

Comment:
this comment would be better as a docstring comment for the function https://peps.python.org/pep-0257/#one-line-docstrings since it relates more to the behaviour of the function rather than the context where it is called